src/notebooks/controllers/kernelSource/notebookKernelSourceSelector.ts

```diff
@@ -534,23 +648,61 @@ export class NotebookKernelSourceSelector implements INotebookKernelSourceSelect
                 }
             }
         });
-        if (provider.status === 'discovering') {
-            quickPick.busy = true;
-        }
+        // Assume we're always busy.
+        quickPick.busy = true;
+        let noLongerBusy = true;
+        // If we don't get any updates after 2s, then hide the busy indicator
+        let updateTimeout = setTimeout(() => {
+            if (noLongerBusy) {
+                quickPick.busy = false;
+            }
+        }, 2_000);
+        state.disposables.push(new Disposable(() => clearTimeout(updateTimeout)));
         provider.onDidChangeStatus(
             () => {
+                clearTimeout(updateTimeout);
                 switch (provider.status) {
                     case 'discovering':
                         quickPick.busy = true;
+                        noLongerBusy = false;
                         break;
                     case 'idle':
-                        quickPick.busy = false;
+                        noLongerBusy = true;
+                        updateTimeout = setTimeout(() => {
+                            if (noLongerBusy) {
+                                quickPick.busy = false;
+                            }
+                        }, 2_000);
+                        state.disposables.push(new Disposable(() => clearTimeout(updateTimeout)));
                         break;
                 }
             },
             this,
             state.disposables
         );
+
+        const recommendedItems: (QuickPickItem | ConnectionQuickPickItem)[] = [];
+        const updateRecommended = () => {
+            if (!provider.recommended) {
+                return;
+            }
+            if (!recommendedItems.length) {
+                recommendedItems.push(<QuickPickItem>{
+                    label: DataScience.recommendedKernelCategoryInQuickPick(),
+                    kind: QuickPickItemKind.Separator
+                });
+            }
+            const recommendedItem = connectionToQuickPick(provider.recommended);
+            recommendedItem.label = `$(star-full) ${recommendedItem.label}`;
+            if (recommendedItems.length === 2) {
+                recommendedItems[1] = recommendedItem;
+            } else {
+                recommendedItems.push(recommendedItem);
+            }
+            updateKernelQuickPickWithNewItems(quickPick, recommendedItems.concat(quickPickItems), recommendedItems[1]);
```

Comment:
Do we want to use `updateKernelSourceQuickPickWithNewItems` here too?

Reply:
Thats was a wrong function, not re-usable.